src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -657,6 +649,9 @@ private static IEnumerable<TSource> TakeWhileIterator<TSource>(IEnumerable<TSour
         public static IEnumerable<TSource> Skip<TSource>(this IEnumerable<TSource> source, int count)
         {
             if (source == null) throw Error.ArgumentNull("source");
+            if (count < 0) count = 0;
```

Comment:
Yep, that's pretty much my thinking as to "objections in term of exposed interfaces" which was a thinko for "objections in terms of exposed objects".